### PR TITLE
feat(preprocessEvent): emit preprocessEvent for profiles 

### PR DESCRIPTION
**Pull request by JonasBa**
Implements preprocessEvent hook implementation for profiles (counterpart to the replay [PR](https://github.com/getsentry/sentry-javascript/pull/9034/files#diff-132075a904d758e2c4ac08f182e4c6e91a1edb58766aed48fbb44e45730b075f))

### CODE REVIEW

File: src/integration.ts [hunk 1]
```diff
@@ -143,6 +143,11 @@ export class ProfilingIntegration implements Integration {
           PROFILE_QUEUE.splice(profileIndex, 1);
           const profile = createProfilingEvent(cpuProfile, profiledTransaction);
 
+          if (client.emit) {
+            // @ts-expect-error preprocessEvent hook doesn't exist?
```
Comment (mydea) [inline src/integration.ts:147]: Weird, any idea why this does not exist here? Should be on both the client and Badeclient types 🤔

**Review by mydea — APPROVED** [review]

File: src/integration.ts (hunk 1 above)
Comment (JonasBa) [reply src/integration.ts:147]: I'll check, thought it might be because the changes are not in the latest release yet

Comment (JonasBa) [reply src/integration.ts]: @mydea seems like it was just ts being cryptic due to the unknown Event.

File: src/integration.ts [hunk 2]
```diff
@@ -143,6 +143,25 @@ export class ProfilingIntegration implements Integration {
           PROFILE_QUEUE.splice(profileIndex, 1);
           const profile = createProfilingEvent(cpuProfile, profiledTransaction);
 
+          if (client.emit && profile) {
+            const integrations =
+              // @ts-expect-error _integrations is private
+              typeof client._integrations === 'object' &&
```
Comment (mydea) [inline src/integration.ts:149]: Just a tip: you can use `client['_integrations']` to access private fields without TS complaining.

**Pull request merged**